### PR TITLE
adding .src to hash.get results

### PR DESCRIPTION
**Pull request by rf**
Adds `.src` on `RouteResult`s for easier instrumenting

cc @Raynos @sh1mmer

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ function HttpHash() {
         return new HttpHash();
     }
 
-    this._hash = new RouteNode();
+    this._hash = new RouteNode(null, null, false, "/");
```
Comment (Raynos) [inline index.js]: style: single quotes.

**Comment by Raynos** [comment]
lgtm.

**Comment by Matt-Esch** [comment]
I think this is the wrong way to go for introspection because many routes map to one handler. Even suggesting `/` is the root is incorrect. I suggest we decorate the node with the entire pattern used in the set as and when we set the handler function on the node. That way we have some knowledge of the original pattern and not some reconstruction after splitting the route.

**Comment by rf** [comment]
I'm not sure I understand what you mean. If you're building a tree of nodes, each one of which can have a handler, each node has to have knowledge of the path to that node (unless it's rebuilt in `get`, which is silly). 

It doesn't really matter if many routes map to one handler; what I care about is getting the pattern that actually matched.

**Comment by Raynos** [comment]
can you not just do `hash.pathname = pathname;` on line 109 ?

**Comment by rf** [comment]
Yes

File: index.js [hunk 2]
```diff
@@ -105,6 +105,8 @@ function set(pathname, handler) {
         }
     }
 
+    hash.src = pathname;
+
     if (!hash.handler) {
         hash.handler = handler;
```
Comment (Matt-Esch) [inline index.js:110]: You should really be defining hash.src inside the if statement, just in case someone believes they can recover from the exception...

**Comment by Raynos** [comment]
lgtm. #shipit

**Pull request merged**